File: Languages/Java/Transaction.java
package Languages.Java;

// what is the abstract class mean ?? 
public abstract class Transaction {
    private int accountNumber; 
    private Screen screen; 
    private BankDatabase bankDatabase;

    public Transaction(int userAccountNumber, Screen atmScreen, BankDatabase atmBankDatabase) {
        accountNumber = userAccountNumber;
        screen = atmScreen;
        bankDatabase = atmBankDatabase;
    }

    // return account number 
    public int getAccountNumber() {
        return accountNumber; 
    }

    // return reference to screen 
    public Screen getScreen() {
        return screen; 
    }

    // return reference to bank database 
    public BankDatabase getBankDatabase() {
        return bankDatabase; 
    }

    // perform the transaction which will be overriden by subclasses 
    abstract public void execute();

    
}

File: Languages/Golang/Transaction.go
package main

// is there no abstract in GOLANG??
type Transaction struct {
	accountNumber int
	screen        *Screen
	bankDatabase  *BankDatabase
}

func NewTransaction(userAccountNumber int, atmScreen *Screen, atmBankDatabase *BankDatabase) *Transaction {
	return &Transaction{
		accountNumber: userAccountNumber,
		screen:        atmScreen,
		bankDatabase:  atmBankDatabase,
	}
}

func (t *Transaction) getAccountNumber() int {
	return t.accountNumber
}

func (t *Transaction) getScreen() *Screen {
	return t.screen
}

func (t *Transaction) getBankDatabase() *BankDatabase {
	return t.bankDatabase
}

func (t *Transaction) Execute() {
	// implementation here
}
